### ISSUE TITLE
Convert hand relative rotation to global rotation

### ISSUE THREAD

**Issue by tfederico** [facebookresearch/frankmocap#181]
Hello,

how do you convert the hand rotations (which according to the documentation are relative rotations) to global?

Also is the order of the angles xyz or zyx? I would like to convert them to quaternions

**Comment by penincillin** [comment]
@tfederico 
You can use the function [transfer_rotation](https://github.com/facebookresearch/frankmocap/blob/1c0f78907e82f95e183ecae20f6a70226c7c7a18/integration/copy_and_paste.py#L41) to convert the hand rotation from relative to global.
The order of angles is ```xyz```.

**Comment by tfederico** [comment]
What's the value of `part_idx` parameter for the fingers joints? By looking at the code, it seems that the index refers to the body joints

**Comment by penincillin** [comment]
The ```part_idx``` of left hand and right hand joints range from ```25``` to ```39``` for left hands and ```40``` to ```54``` for right hands. The detailed order of each hand joint (same for both the left and right hand) can be found [here](https://github.com/facebookresearch/frankmocap/blob/main/docs/joint_order.md#joint-angle-hand).

**Comment by tfederico** [comment]
So what I tried to do was:

```
from demo.demo_options import DemoOptions
from bodymocap.body_mocap_api import BodyMocap
import torch
from integration.copy_and_paste import transfer_rotation
import pickle as pkl
import numpy as np


def main():
    args = DemoOptions().parse()
    args.use_smplx = True

    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
    assert torch.cuda.is_available(), "Current version only supports GPU"

    # Set Mocap regressor
    body_mocap = BodyMocap(args.checkpoint_body_smplx, args.smpl_dir, device=device, use_smplx=True)

    with open("mocap_output/mocap/00000_prediction_result.pkl", "rb") as fp:
        data = pkl.load(fp)
    pred_output_list = data['pred_output_list'][0]
    pred_body_pose = pred_output_list['pred_body_pose']
    pred_left_hand_pose = pred_output_list['pred_left_hand_pose'][0]

    for part_idx in range(15):
        part_rot = np.array([pred_left_hand_pose[part_idx*3:(part_idx+1)*3]])
        glob_rot = transfer_rotation(body_mocap.smpl, pred_body_pose, part_rot, 25+part_idx, transfer_type="l2g", result_format="rotmat")
```

but I got an error from [here](https://github.com/facebookresearch/frankmocap/blob/1c0f78907e82f95e183ecae20f6a70226c7c7a18/integration/copy_and_paste.py#L62) because `body_pose` has 72 elements instead of 66, so I replaced the `(22, 3)` with `(24,3)`. Then I got another error [here](https://github.com/facebookresearch/frankmocap/blob/1c0f78907e82f95e183ecae20f6a70226c7c7a18/integration/copy_and_paste.py#L28), specifically `IndexError: index 25 is out of bounds for dimension 0 with size 24`.

Did I do anything wrong? My guess is that it's using the body indices, which is why it cannot find index 25.

**Comment by tfederico** [comment]
Shall I be using the `hand_mocap` instead of `body_mocap`?
If so, when I call `transfer_rotation` what shall I pass instead of `pred_body_pose`?

**Comment by tfederico** [comment]
I think I figured it out, by concatenating modifying `pred_body_pose` to
```
pred_body_pose = np.concatenate((pred_body_pose, pred_output_list['pred_left_hand_pose'], pred_output_list['pred_right_hand_pose']), axis=1)
```
and changing [these 2 lines](https://github.com/facebookresearch/frankmocap/blob/1c0f78907e82f95e183ecae20f6a70226c7c7a18/integration/copy_and_paste.py#L61-L62) I was able to make it run. In this way it's building the kinematic tree using also the hands.

Am I right to assume this is the correct approach?
